open_spiel/algorithms/infostate_tree.cc

```diff
@@ -491,12 +491,57 @@ std::shared_ptr<InfostateTree> MakeInfostateTree(
     const Game& game, Player acting_player,
     int max_move_limit) {
   return std::shared_ptr<InfostateTree>(new InfostateTree(
```

Comment:
Use `std::make_shared`, it shortens code like this. Also, it would allow you to search for the `new` keyword to check if everything uses smart pointers (skipping the low level parts that might need it).

Reply:
Had to do it this way because of friendliness.

Reply:
Aha, that is a good reason to do it the old way.